### ISSUE TITLE
In VSCode Commands are run before the UI is available

### ISSUE THREAD

**Issue by smacintyre** [gitpod-io/gitpod#4220]
### Bug description

This was not an issue in Theia; this used to work fine.

If you have a command in your `.gitpod.yml` that requires the UI, for example it it needs to open the browser, the command will be executed before the UI is available. 

Practical use case. Using AWS SSO needs to open a browser to create temporary credentials. This command needs to be run when a workspace opened. In the example repo below, this worked fine in Theia. However in VSCode, the command is run before the UI is available so it tries to open the URL in `lynx`. Which doesn't work.

### Steps to reproduce

Try opening https://github.com/selfstructured/science

You will see `lynx` browser instead of the VSCode browser.

This is the case for any command that must open a web page.

### Expected behavior

The commands in .gitpod.yml should only be run once the UI is available or there should be some way to wait for it to be available before running. 

### Example repository

See: https://github.com/selfstructured/science

### Anything else?

For now I have a workaround where I just wait for any key to be pressed before starting the script.

Sorry for the wait filing this, this issue has been present since the cut over to VSCode. I've just been to busy to create an example repo.

**Comment by akosyakov** [comment]
duplicate of https://github.com/gitpod-io/gitpod/issues/4036
